tests/python_package_test/test_basic.py

```diff
@@ -609,3 +610,101 @@ def test_custom_objective_safety():
     good_bst_multi.update(fobj=_good_gradients)
     with pytest.raises(ValueError, match=re.escape(f"number of models per one iteration ({nclass})")):
         bad_bst_multi.update(fobj=_bad_gradients)
+
+
+def test_numpy_regular_dtypes():
```

Comment:
```suggestion
def test_pandas_with_numpy_regular_dtypes():
```

Also, please move this and `test_pandas_nullable_dtypes` tests to `test_engine.py` file because `lgb.train()` function that is defined in `engine.py` is used in these tests.

Reply:
addressed in 8bf1617